vita3k/renderer/src/gl/renderer.cpp

```diff
@@ -482,19 +482,29 @@ void get_surface_data(GLContext &context, size_t width, size_t height, size_t st
     }
 
     glPixelStorei(GL_PACK_ROW_LENGTH, static_cast<GLint>(stride_in_pixels));
-    if (format == SCE_GXM_COLOR_FORMAT_U8U8U8U8_ABGR) {
+    switch (format) {
+    case SCE_GXM_COLOR_FORMAT_U8U8U8U8_ABGR:
+    case SCE_GXM_COLOR_FORMAT_U8U8U8U8_ARGB:
         glReadPixels(0, 0, static_cast<GLsizei>(width), static_cast<GLsizei>(height), GL_RGBA, GL_UNSIGNED_BYTE, pixels);
-    } else if (format == SCE_GXM_COLOR_FORMAT_U8U8U8U8_RGBA) {
+        break;
+    case SCE_GXM_COLOR_FORMAT_U8U8U8U8_RGBA:
         glReadPixels(0, 0, static_cast<GLsizei>(width), static_cast<GLsizei>(height), GL_RGBA, GL_UNSIGNED_BYTE, pixels);
-        for (int i = 0; i < width * height; ++i) {
+        for (int i = 0; i < width * height; i++) {
             uint8_t *pixel = reinterpret_cast<uint8_t *>(&pixels[i]);
             std::swap(pixel[0], pixel[3]);
             std::swap(pixel[1], pixel[2]);
         }
-    } else {
-        // TODO wise color conversino implementation
-        // maybe we can use PBO?
-        assert(false);
+        break;
+    case SCE_GXM_COLOR_FORMAT_U8U8U8_BGR:
```

Comment:
I saw that this is working, but it's really weird. (we should swap some colors in theory) We need to investigate this further.

Reply:
No really strange, abgr and argb works for rgba

Reply:
Im going to do it.

Reply:
Then also remove swap in rgba. Otherwise it makes no sense.

Reply:
yes i have no understund this swap, but you have write it for what game ?